tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -38,120 +38,144 @@ describe(__filename, () => {
-        slug,
+      const forAddonSlug = 'addon-slug';
+      const state = reducer(undefined, loadAddonsByAuthors({
+        forAddonSlug,
         // This is the case where there are more add-ons loaded than needed.
-        addons: Array(OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE + 2).fill(fakeAddon),
+        addons: Array(ADDONS_BY_AUTHORS_PAGE_SIZE + 2).fill(fakeAddon),
       }));
-      expect(state.byAddonSlug[slug])
-        .toHaveLength(OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE);
+      expect(state.byAddonSlug[forAddonSlug])
+        .toHaveLength(ADDONS_BY_AUTHORS_PAGE_SIZE);
+    });
+
+    it('returns state if no excluded slug is specified', () => {
+      const forAddonSlug = 'addon-slug';
+
+      const previousState = reducer(undefined, loadAddonsByAuthors({
+        addons: [{ ...fakeAddon, id: 3 }],
+        forAddonSlug,
+      }));
+      expect(previousState.byAddonSlug)
+        .toEqual({ 'addon-slug': [3] });
+
+      const state = reducer(previousState, fetchAddonsByAuthors({
+        authors: ['author2'],
+        addonType: ADDON_TYPE_THEME,
+        errorHandlerId: 'error-handler-id',
+      }));
+
+      expect(state.byAddonSlug).toEqual({ 'addon-slug': [3] });
     });
 
     it('resets the loaded add-ons', () => {
-      const slug = 'addon-slug';
+      const forAddonSlug = 'addon-slug';
 
-      const previousState = reducer(undefined, loadOtherAddonsByAuthors({
-        addons: [fakeAddon],
-        slug,
+      const previousState = reducer(undefined, loadAddonsByAuthors({
+        addons: [{ ...fakeAddon, id: 4 }],
+        forAddonSlug,
       }));
       expect(previousState.byAddonSlug)
-        .toEqual({ 'addon-slug': [createInternalAddon(fakeAddon)] });
+        .toEqual({ 'addon-slug': [4] });
 
-      const state = reducer(previousState, fetchOtherAddonsByAuthors({
+      const state = reducer(previousState, fetchAddonsByAuthors({
         authors: ['author1'],
         addonType: ADDON_TYPE_THEME,
         errorHandlerId: 'error-handler-id',
-        slug,
+        forAddonSlug,
       }));
-      expect(state.byAddonSlug)
-        .toEqual({ 'addon-slug': undefined });
+
+      expect(state.byAddonSlug).toEqual({ 'addon-slug': undefined });
     });
   });
 
-  describe('fetchOtherAddonsByAuthors()', () => {
-    const getParams = () => {
+  describe('loadAddonsByAuthors()', () => {
+    const getParams = (extra = {}) => {
       return {
-        authors: ['user1', 'user2'],
-        addonType: ADDON_TYPE_THEME,
-        errorHandlerId: 'error-handler-id',
-        slug: 'addon-slug',
+        addons: [],
+        forAddonSlug: fakeAddon.slug,
+        ...extra,
       };
     };
 
-    it('requires an error id', () => {
-      const params = getParams();
-      delete params.errorHandlerId;
-      expect(() => {
-        fetchOtherAddonsByAuthors(params);
-      }).toThrow(/An errorHandlerId is required/);
-    });
+    it('adds each add-on to each author array', () => {
```

Comment:
This test and the next test fail with seemingly extra add-ons/authors added to data. Their ids match the defaults in `fakeAddon` and `fakeAuthor` but I'm just totally stumped as to what's happening. I think I need a second pair of eyes 👀

Reply:
This fixes it:

```diff
diff --git a/src/amo/reducers/addonsByAuthors.js b/src/amo/reducers/addonsByAuthors.js
index d7cfa3661..a9c26c691 100644
--- a/src/amo/reducers/addonsByAuthors.js
+++ b/src/amo/reducers/addonsByAuthors.js
@@ -1,4 +1,5 @@
 /* @flow */
+import deepcopy from 'deepcopy';
 import invariant from 'invariant';
 
 import { createInternalAddon } from 'core/reducers/addons';
@@ -102,7 +103,7 @@ const reducer = (
         },
       };
     case LOAD_ADDONS_BY_AUTHORS: {
-      const newState = { ...initialState };
+      const newState = deepcopy(initialState);
 
       if (action.payload.forAddonSlug) {
         newState.byAddonSlug = {
```

The way I found out is by typing `t` (equivalent of [`-t`](https://facebook.github.io/jest/docs/en/cli.html)) and entering a regex to only run each test in isolation. They pass in isolation which made me wonder about the deep object references -- that's where the problem is.

Reply:
Ironically, I think `deepcopy` had bitrotted as a dep. We still have it in `package.json` but I can't find anything that uses it.

Reply:
Software is hard.

Thanks a bunch for looking into this, I figured it was something like that as I noticed commenting out certain tests would change the outcome of others as well, but I was just getting nowhere with it. I owe you a 🍺 for this 🕵️ work.